lms/djangoapps/course_goals/models.py

```diff
@@ -84,3 +90,41 @@ class Meta:
     user = models.ForeignKey(User, on_delete=models.CASCADE)
     course_key = CourseKeyField(max_length=255)
     date = models.DateField()
+
+    @classmethod
+    def populate_user_activity(cls, user, course_key, request=None, check_if_mobile_app=False):
```

Comment:
1. sure will update the name
2. i wasn't passing the request for the one web spot, and all the mobile code will need to be removed soon, so at that point i think we can remove the request variable (i think better to also allow passing the user to avoid any masquerade issues)

Reply:
changed to only_if_mobile_app

Reply:
I think as you add masquerade checks, you'd want the request object anyway.

Reply:
i've added the is_masquerade check, i don't think i need the request object for that?

Reply:
Ah right - sorry. In my head the masquerade code went and got the current request from crum. It just grabs the masquerade settings from the user object.

It still feels cleaner to me to just require the request object, and then grab the user from there. This method is always in the context of a request because it is tracking user activity, which come in the form of requests. And it has logic inside that wants to check that request.

I know that that logic (is mobile request) only happens (and thus request is only required) if `only_if_mobile_app` is `True` - but that's an implicit relationship between arguments that I don't love.

So to me, it feels cleaner to say "I know the caller has a request. I kind of want to ensure the call has a request, actually. I can grab user and mobile info from that. And it removes implicit relationships between arguments."

Whereas maybe you're thinking "I like that the user argument is explicit, not implicit. And I don't want callers that don't care about mobile to need to handle the request object." Sound right?

If you are unpursuaded, can you at least make the implicit relationship between arguments clear in the function's docstring?

Reply:
i think my argument was that once we remove the mobile calls in the near future, we can remove the mobile argument and then it wouldn't make sense to pass the request only for the purposes of getting the user
i'd agree with you if i thought the mobile argument was going to stick around
but i can add documentation about the implicit relationship